src/template_sql/cdc_sql_template.sql

```diff
@@ -23,12 +23,10 @@ USING (
     ),
     -- To handle occasional dups from SLT connector
     S1 AS (
-      SELECT * EXCEPT(row_num)
-      FROM (
-        SELECT *, ROW_NUMBER() OVER (PARTITION BY ${keys}, recordstamp ORDER BY recordstamp) AS row_num
-        FROM S0
-      )
-      WHERE row_num = 1
+      SELECT r.* FROM (
+                      SELECT ${keys}, ARRAY_AGG(S0 ORDER BY S0.recordstamp DESC LIMIT 1)[OFFSET(0)] r
```

Comment:
Thanks for the PR, and apologies for delay in reviewing this.

The intent in this subquery is to handle duplicates - including `recordstamp`. Because of that, you will also need `recordstamp` along with the `${keys}` here (in `SELECT` as well as `GROUP BY`).

Also, I did a performance comparison.
For the first run (when CDC table is empty), this method has sever negative impact for tall and wide tables - the elapsed time goes from seconds to minutes, and slot time is 25-30 times higher too for a table like vbak.

Yes, only the initial runs have to deal with so many rows, and that subsequent runs should not have many records to process. But at this point, we will continue with the current approach until we have time to compare on different sizes of tables (tall/narrow, short/wide etc) and see if we can see a gain here. Logically, your approach makes sense, and I was expecting it to be better in terms of performance, but that's not the case apparently.